cmake/SpectreAddTestLibs.cmake

```diff
@@ -91,10 +91,9 @@ function(add_test_library LIBRARY FOLDER LIBRARY_SOURCES LINK_LIBS)
 
   target_link_libraries(
     ${LIBRARY}
-    INTERFACE
-    ${LINK_LIBS}
-    INTERFACE
-    ${PYTHON_LIBRARIES}
+    PRIVATE
+      ${LINK_LIBS}
```

Comment:
General comment, please don't indent these more than what `PRIVATE`,etc. are already indented

Reply:
do you really prefer that? Perhaps we should add this indentation while updating the linking?

Reply:
Yes, no extra indentation is consistent with how we've been doing all of our CMake formatting. No, I don't want the extra spaces. Currently we've been using the formatting Emacs provides, which is not the extra indentation.

Reply:
fine with me. rebased an squashed.

Reply:
Just a ping on this file :) The rest look good!